<launch>

    <arg name="image" default="/camera/color/image_raw"/>

    <!-- DETECTNET -->
	<arg name="model_name" default="ssd-mobilenet-v2"/>
	<arg name="model_path" default=""/>
	<arg name="prototxt_path" default=""/>
	<arg name="class_labels_path" default=""/>
	<arg name="input_blob" default=""/>
	<arg name="output_cvg" default=""/>
	<arg name="output_bbox" default=""/>
	<arg name="overlay_flags" default="box,labels,conf"/>
	<arg name="mean_pixel_value" default="0.0"/>
	<arg name="threshold" default="0.5"/>

	<include file="$(find yocs_velocity_smoother)/launch/velocity_smoother_with_params.launch"/>

	<!-- run detect_leg_clusters -->  
	<param name="forest_file" value="$(find leg_tracker)/config/trained_leg_detector_res=0.33.yaml" />
	<param name="scan_topic" value="/scan" />
	<param name="fixed_frame" value="base_link"/>
	<param name="scan_frequency" value="15"/>
	<node pkg="leg_tracker" type="detect_leg_clusters" name="detect_leg_clusters" output="screen"/>  
    
    <node name="cogniteam_person_follower_node" pkg="cogniteam_person_follower" type="cogniteam_person_follower_node" output="screen">
        <!-- <rosparam file="$(find person_3d_locator)/param/person_3d_locator_param.yaml" /> -->
        <param name="color_image_topic" type="string" value="$(arg image)"/>
        <param name="model_name" value="$(arg model_name)"/>
		<param name="model_path" value="$(arg model_path)"/>
		<param name="prototxt_path" value="$(arg prototxt_path)"/>
		<param name="class_labels_path" value="$(arg class_labels_path)"/>
		<param name="input_blob" value="$(arg input_blob)"/>
		<param name="output_cvg" value="$(arg output_cvg)"/>
		<param name="output_bbox" value="$(arg output_bbox)"/>
		<param name="overlay_flags" value="$(arg overlay_flags)"/>
		<param name="mean_pixel_value" value="$(arg mean_pixel_value)"/>
		<param name="threshold" value="$(arg threshold)"/>
    </node>
    <!-- remap from="/person_3d_locator/nearest_person" to="/move_base_simple/goal"/-->
</launch>
